tests/filecheck/dialects/memref/invalid_ops.mlir

```diff
@@ -35,3 +35,13 @@ builtin.module {
 }) {function_type = () -> (), sym_name = "invalid_reassociation"} : () -> ()
 
 // CHECK: Expected attribute i64
+
+// -----
+
+"func.func"() ({
+  %0 = "memref.alloc"() <{operandSegmentSizes = array<i32: 0, 0>}> : () -> memref<10x2xindex>
+  %1 = "memref.reinterpret_cast"(%0) <{operandSegmentSizes = array<i32: 1, 0, 0, 0>, static_offsets = array<i64: 0>, static_sizes = array<i64: 5, 4>, static_strides = array<i64: 1, 1>}> : (memref<10x2xindex>) -> memref<10x2xindex, strided<[1, 1]>>
+  "func.return"() : () -> ()
+}) {function_type = () -> (), sym_name = "mismatched_sizes"} : () -> ()
+
+// CHECK: Expected result type with size = 5 instead of 10 in dim = 0
```

Comment:
There are three kinds of verification errors you added in the `verify_` below, can you please add a test case for each of these?